authalligator_client/entities.py

```diff
@@ -196,68 +192,68 @@ class DeleteAccountPayload(DeleteOperation):
 class AuthorizeAccountPayload(BaseAAEntity):
     TYPENAME = "AuthorizeAccountPayload"
 
-    account = attr.attrib(
-        converter=entity_converter(Account),  # type: ignore[misc]
-    )  # type: Account
-    account_key = attr.attrib()  # type: str
-    number_of_account_keys = attr.attrib()  # type: int
+    account: Account = attr.attrib(
+        converter=entity_converter(Account)  # type: ignore[misc],
+    )
+    account_key: str = attr.attrib()
+    number_of_account_keys: int = attr.attrib()
 
 
 @attr.attrs(frozen=True)
 class VerifyAccountPayload(BaseAAEntity):
     TYPENAME = "VerifyAccountPayload"
 
-    account = attr.attrib(
+    account: Account = attr.attrib(
         converter=entity_converter(Account),  # type: ignore[misc]
-    )  # type: Account
+    )
 
 
 @attr.attrs(frozen=True)
 class Query(BaseAAEntity):
-    account = attr.attrib(
+    account: Union[Omitted, Account, AccountError] = attr.attrib(
         default=OMITTED,
-        converter=entity_converter([Account, AccountError]),  # type: ignore[misc]
-    )  # type: Union[Omitted, Account, AccountError]
+        converter=entity_converter([Account, AccountError])  # type: ignore[misc]
+    )
 
 
 @attr.attrs(frozen=True)
 class Mutation(BaseAAEntity):
     # mypy and the attrs plugin doens't like the `Omitted` default + converter
     # stuff
-    authorize_account = attr.attrib(  # type: ignore
+    authorize_account: Union[Omitted, AuthorizeAccountPayload, AccountError] = attr.attrib(  # type: ignore
```

Comment:
This comment applies to here and all the other places below:

Do we need that `type: ignore` at the end of this line? Like, what's the point of adding typing information if it's then ignored?

Reply:
We're ignoring field type vs converter mismatch. The field type is still available for all other code.

Reply:
`type: ignore` is in the wrong place then, right? It should be on the `converted=` line.

Reply:
I tried removing it but it would trigger this, but it's indeed about the return type of `attrib`. This code just pushes further than what Mypy can understand.

```
authalligator_client/entities.py:226: error: No overload variant of "attrib" matches argument types "Omitted", "Union[Omitted, AuthorizeAccountPayload, AccountError]"  [call-overload]
authalligator_client/entities.py:226: note: Possible overload variants:
authalligator_client/entities.py:226: note:     def attrib(default: None = ..., validator: None = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: None = ..., converter: None = ..., factory: None = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:226: note:     def [_T] attrib(default: None = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:226: note:     def [_T] attrib(default: _T, validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:226: note:     def [_T] attrib(default: Optional[_T] = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: object = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:234: error: No overload variant of "attrib" matches argument types "Omitted", "Union[Omitted, VerifyAccountPayload, AccountError]"  [call-overload]
authalligator_client/entities.py:234: note: Possible overload variants:
authalligator_client/entities.py:234: note:     def attrib(default: None = ..., validator: None = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: None = ..., converter: None = ..., factory: None = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:234: note:     def [_T] attrib(default: None = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:234: note:     def [_T] attrib(default: _T, validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:234: note:     def [_T] attrib(default: Optional[_T] = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: object = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:241: error: No overload variant of "attrib" matches argument types "Omitted", "Union[Omitted, DeleteAccountPayload, AccountError]"  [call-overload]
authalligator_client/entities.py:241: note: Possible overload variants:
authalligator_client/entities.py:241: note:     def attrib(default: None = ..., validator: None = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: None = ..., converter: None = ..., factory: None = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:241: note:     def [_T] attrib(default: None = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:241: note:     def [_T] attrib(default: _T, validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:241: note:     def [_T] attrib(default: Optional[_T] = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: object = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:250: error: No overload variant of "attrib" matches argument types "Omitted", "Union[Omitted, DeleteAccountKeyPayload, AccountError]"  [call-overload]
authalligator_client/entities.py:250: note: Possible overload variants:
authalligator_client/entities.py:250: note:     def attrib(default: None = ..., validator: None = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: None = ..., converter: None = ..., factory: None = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:250: note:     def [_T] attrib(default: None = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:250: note:     def [_T] attrib(default: _T, validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:250: note:     def [_T] attrib(default: Optional[_T] = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: object = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:259: error: No overload variant of "attrib" matches argument types "Omitted", "Union[Omitted, DeleteOtherAccountKeysPayload, AccountError]"  [call-overload]
authalligator_client/entities.py:259: note: Possible overload variants:
authalligator_client/entities.py:259: note:     def attrib(default: None = ..., validator: None = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: None = ..., converter: None = ..., factory: None = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
authalligator_client/entities.py:259: note:     def [_T] attrib(default: None = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:259: note:     def [_T] attrib(default: _T, validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: Optional[Type[_T]] = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> _T
authalligator_client/entities.py:259: note:     def [_T] attrib(default: Optional[_T] = ..., validator: Union[Callable[[Any, Attribute[_T], _T], Any], Sequence[Callable[[Any, Attribute[_T], _T], Any]], None] = ..., repr: Union[bool, Callable[[Any], str]] = ..., cmp: Union[bool, Callable[[Any], Any], None] = ..., hash: Optional[bool] = ..., init: bool = ..., metadata: Optional[Mapping[Any, Any]] = ..., type: object = ..., converter: Optional[Callable[[Any], Any]] = ..., factory: Optional[Callable[[], _T]] = ..., kw_only: bool = ..., eq: Union[bool, Callable[[Any], Any], None] = ..., order: Union[bool, Callable[[Any], Any], None] = ..., on_setattr: Union[Callable[[Any, Attribute[Any], Any], Any], List[Callable[[Any, Attribute[Any], Any], Any]], _NoOpType, None] = ..., alias: Optional[str] = ...) -> Any
Found 5 errors in 1 file (checked 10 source files)
```